tests/test_scheduler.py

```diff
@@ -282,8 +282,11 @@ def test_full_loop_no_noise(self):
         result_sum = torch.sum(torch.abs(sample))
         result_mean = torch.mean(torch.abs(sample))
 
-        assert abs(result_sum.item() - 259.0883) < 1e-2
-        assert abs(result_mean.item() - 0.3374) < 1e-3
+        # Large tolerance because of differences on GPU/CPU
```

Comment:
Hmm not very happy about this. Note that they shouldn't be that much of a difference because the model doesn't do anything - it's a dummy model and not run on GPU. It'd be nice if we could revert the change here. I think what's happening here is that we don't pass a generator to `step(...)` note that DDPM is stochastic and therefore different hardware gives different results. We can solve this simply by passing a generator with `generator = torch.manual_seed(0)`